### ISSUE TITLE
AssertionError: shift operations

### ISSUE THREAD

**Issue by jserv** [jserv/amacc#102]
The following error occurred when `scripts/runtest.py` was being run.
```
FAIL: test_shift (__main__.TestCC_SC)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "scripts/runtest.py", line 54, in test
    self.assertEqual(amacc_out.decode('utf-8'), gcc_out.decode('utf-8'))
AssertionError: '1 <<[70 chars]-1 = 2\n4 >> -1 = 8\n-1 << 1 = fffffffe\n-1 <<[210 chars]68\n' != '1 <<[70 chars]-1 = 0\n4 >> -1 = 0\n-1 << 1 = fffffffe\n-1 <<[210 chars]68\n'
  1 << 0 = 1
  1 << 2 = 4
  0 << 4 = 0
  1 << 31 = 80000000
  1 << 32 = 0
- 4 << -1 = 2
?           ^
+ 4 << -1 = 0
?           ^
- 4 >> -1 = 8
?           ^
+ 4 >> -1 = 0
?           ^
  -1 << 1 = fffffffe
  -1 << 0 = ffffffff
  4 >> 1 = 2
  4 >> 5 = 0
  0x80000000 >> 31 = ffffffff
  -1 >> 2 = ffffffff
  -1091119768 - -1091119766 = -2(fffffffe)
  a = f6d5680c, b = bef6d568, c = fff6d568
  a = 76d5680c, b = bef6d568, c = bef6d568
```

**Comment by HPCguy** [comment]
Sorry. I have been enjoying the holiday here in the US and just saw this before sleeping tonight. I will fix it first thing tomorrow.

**Comment by vacantron** [comment]
@HPCguy I think you should use the cross-compiler mentioned in prerequisites.

Use negative amount in shift operation is an undefined behavior in C standard. GCC seems to have different implementation on different target.

**Comment by HPCguy** [comment]
@vacantron,
Thank you for resolving this.  The negative shift feature was implemented by splasky in pull request #47, fixing issue #39.  I was unaware that the negative shift was implementation specific in the C language, and unfortunately did not have the QEMU environment to follow up on this and track this down.
-HPCguy
